server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -113,7 +137,8 @@ private SearchHits buildSearchHits(SearchContext context, Profiler profiler) {
         NestedDocuments nestedDocuments = context.getSearchExecutionContext().getNestedDocuments();
 
         PreloadedSourceProvider sourceProvider = new PreloadedSourceProvider();
-        context.getSearchExecutionContext().setSourceProvider(sourceProvider);
+        PreloadedFieldLookupProvider fieldLookupProvider = new PreloadedFieldLookupProvider();
+        context.getSearchExecutionContext().setLookupProviders(sourceProvider, ctx -> fieldLookupProvider);
```

Comment:
Thinking out load ... Is really required to treat _source different to any other (stored) field here apart from historical reasons?  I guess it is but does it mean we can potentially preload source twice.

Reply:
_source might be synthesised rather than come from a stored field so it needs to be handled separately.  It might be possible to explicitly ask for `_source` via `_fields` which would load things twice, and we can probably intercept that and make sure it doesn't happen.  Will add more tests!

Reply:
We already have a check for `_source` in stored fields, which gets ignored.